beeline/src/java/org/apache/hive/beeline/BeeLine.java

```diff
@@ -1359,22 +1359,53 @@ private int executeFile(String fileName) {
     }
   }
 
+  public String executeReader(ConsoleReader reader, Character mask) {
+    String line = null;
+    StringBuilder qsb = new StringBuilder();
+    try {
+      while ((line = (getOpts().isSilent() && getOpts().getScriptFile() != null) ? reader
+              .readLine(null, mask) : reader.readLine(getPrompt())) != null) {
+        // Skipping through comments
+        if (!line.startsWith("--")) {
+          qsb.append(line + "\n");
+        }
+      }
+      if (qsb.length() > 0) {
+        qsb.delete(qsb.length() - 2, qsb.length());
+      }
+      line = qsb.toString();
+    }
+    catch (Throwable t) {
+      handleException(t);
```

Comment:
Can we consider adding log here with

LOG.error("Unexpected error in thread: {}, message: {}", Thread.currentThread().getName(), t.getMessage());